### ISSUE TITLE
ViewDeestroyedError when click closing notification

### ISSUE THREAD

**Issue by CHBaker** [flauc/angular2-notifications#296]
Hey, I'm getting this error when I click the notification for it to disappear.
```
Uncaught Error: ViewDestroyedError: Attempt to use a destroyed view: detectChanges
    at viewDestroyedError (core.js:9800)
    at Object.debugUpdateDirectives [as updateDirectives] (core.js:14652)
    at checkAndUpdateView (core.js:13802)
    at callWithDebugContext (core.js:15056)
    at Object.debugCheckAndUpdateView [as checkAndUpdateView] (core.js:14593)
    at ViewRef_.detectChanges (core.js:11577)
    at eval (angular2-notifications.umd.js:535) <----------- HERE
    at ZoneDelegate.invoke (zone.js:388)
    at Object.onInvoke (core.js:4749)
    at ZoneDelegate.invoke (zone.js:387)
```
This is very bad for performance as this error hits 365 times in my app because it executes change detection though the entire component tree.
This error is coming from Angular's changed detection running right after it's destroyed, looking for it. I'm not sure how to fix it though, or prevent it maybe in the source code for angular2-notifications

**Comment by pmitrifork** [comment]
I had the same issue today.

**Comment by CHBaker** [comment]
I'd be happy to submit a pull request, I just couldn't find in the source code where I could add the onDestroy hook and what to unsubscribe from

**Comment by Frisky-** [comment]
Any fix for that ?

**Comment by flauc** [comment]
I'm unable to recreate this one. If someone can help set up the recreated error https://stackblitz.com. I can start debugging.

**Comment by CHBaker** [comment]
I can try to recreate it next week, I know it has to do with a subscription that is not unsubscribed from in the source code, It might be caused by lazy loaded routes

**Comment by davidalee** [comment]
I don't know if it's necessarily related to lazy loaded routes as I'm also seeing this error and do not have lazy loaded routes in my app.

I'm not able to reproduce this issue reliably, but when it does fail, it fails spectacularly and crashes the entire app.

**Comment by flauc** [comment]
yes @davidalee you're right as far as I can tell it isn't connected. It looks like it errors out when the notification is manually destroyed at the same time as it's being animated out.

**Comment by tpcsapi** [comment]
I've found out that in my case the similar errors were thrown to console when:

1. I had multiple notifications opened one just after another (f.e. 5 at the same time on the screen),
2. randomly moved mouse cursor over all notifications,
3. pauseOnHover param was set to true.

When I changed pauseOnHover to false the problem disappeared.

Other params:

    {
        timeOut: 3000,
        showProgressBar: true,
        pauseOnHover: true, 
        clickToClose: true,
        animate: 'fromRight'
    }
